format/Schema.fbs

```diff
@@ -252,6 +252,7 @@ table Field {
   // Name is not required, in i.e. a List
   name: string;
   nullable: bool;
+  // This is the type of the index if the field is dictionary encoded
```

Comment:
I don't think this is right. In Java, when a dictionary Field is written it first calls `DictionaryUtility.toMessageFormat` [here](https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/util/DictionaryUtility.java#L45) which changes the Field to be the dictionary type.

Reply:
Ah, thanks, that was the bridge to fix the gap between the java API and the IPC format. If I remember now, people didn't want to embed dictionary-related logic throughout the codebase, which would be required if the internal java field type has the dictionary type.

Reply:
Ok I have checked the integration tests and it seems @BryanCutler is correct, the type here is the type of decoded type, not the index type:

This is taken from the generated json file in integration test 
```
      {
        "name": "dict1_0",
        "type": {
          "name": "utf8"
        },
        "nullable": true,
        "children": [],
        "dictionary": {
          "id": 0,
          "indexType": {
            "name": "int",
            "isSigned": true,
            "bitWidth": 8
          },
          "isOrdered": false
        }
      },
      ...
```

Reply:
@wesm Could you please help confirm what the proper type should be here?